File: src/main/java/io/github/dylmeadows/eontimer/model/ApplicationModel.java
package io.github.dylmeadows.eontimer.model;

import io.github.dylmeadows.eontimer.model.config.ActionConfigurationModel;
import io.github.dylmeadows.eontimer.model.config.TimerConfigurationModel;
import io.github.dylmeadows.eontimer.model.timer.CustomTimerModel;
import io.github.dylmeadows.eontimer.model.timer.Gen3TimerModel;
import io.github.dylmeadows.eontimer.model.timer.Gen4TimerModel;
import io.github.dylmeadows.eontimer.model.timer.Gen5TimerModel;
import lombok.Data;

@Data
public class ApplicationModel {

    private final Gen3TimerModel gen3 = new Gen3TimerModel();
    private final Gen4TimerModel gen4 = new Gen4TimerModel();
    private final Gen5TimerModel gen5 = new Gen5TimerModel();
    private final CustomTimerModel custom = new CustomTimerModel();

    private final ActionConfigurationModel actionSettings = new ActionConfigurationModel();
    private final TimerConfigurationModel timerSettings = new TimerConfigurationModel();
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/ApplicationModel.kt
package io.github.dylmeadows.eontimer.model

import io.github.dylmeadows.eontimer.model.settings.ActionSettingsModel
import io.github.dylmeadows.eontimer.model.settings.TimerSettingsModel
import io.github.dylmeadows.eontimer.model.timer.*
import io.github.dylmeadows.eontimer.util.getValue
import io.github.dylmeadows.eontimer.util.setValue
import javafx.beans.property.ObjectProperty
import javafx.beans.property.SimpleObjectProperty

class ApplicationModel {
    var gen3 = Gen3TimerModel()
    var gen4 = Gen4TimerModel()
    var gen5 = Gen5TimerModel()
    var custom = CustomTimerModel()
    var actionSettings = ActionSettingsModel()
    var timerSettings = TimerSettingsModel()

    val selectedTimerTypeProperty: ObjectProperty<TimerType> = SimpleObjectProperty(TimerConstants.DEFAULT_TIMER_TYPE)
    var selectedTimerType: TimerType by selectedTimerTypeProperty
}